go/vt/vtgate/planbuilder/symtab.go

```diff
@@ -566,7 +566,37 @@ func newResultColumn(expr *sqlparser.AliasedExpr, origin logicalPlan) *resultCol
 		// Just to be safe, generate an anonymous column for the expression.
 		rc.column = &column{
 			origin: origin,
+			typ:    GetReturnType(expr),
 		}
 	}
 	return rc
 }
+
+// GetReturnType returns the type of the select expression that MySQL will return
+func GetReturnType(input sqlparser.SQLNode) querypb.Type {
```

Comment:
the input should be typed to `sqlparser.Expr`. Only expressions can have a return type. `StarExpr`, for example, is not an expression.

Reply:
changed